### DISCUSSION TITLE
Packer File Variables

### DISCUSSION THREAD

**Discussion by kdimail** [actions/virtual-environments#9584] [General questions]
### Breaking changes

Hi there.
I'm stuck with **windows-2019.pkr.hcl** Packer file. Could you please send what exactly Variables and other places do I need to change with my own parameters?
Thanks.

### Target date

Hi there.
I'm stuck with **windows-2019.pkr.hcl** Packer file. Could you please send what exactly Variables and other places do I need to change with my own parameters?
Thanks.

### The motivation for the changes

Hi there.
I'm stuck with **windows-2019.pkr.hcl** Packer file. Could you please send what exactly Variables and other places do I need to change with my own parameters?
Thanks.

### Possible impact

Hi there.
I'm stuck with **windows-2019.pkr.hcl** Packer file. Could you please send what exactly Variables and other places do I need to change with my own parameters?
Thanks.

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [X] Windows Server 2019
- [X] Windows Server 2022

### Mitigation ways

Hi there.
I'm stuck with **windows-2019.pkr.hcl** Packer file. Could you please send what exactly Variables and other places do I need to change with my own parameters?
Thanks.

**Comment by v-sergei-pyshnoi** [comment]
Hello @kdimail . For building runner-images "as is" you do not need modify packer template directly. The most parameters such as `subscription_id, tenant_id etc.` you can pass with args for `packer build` command . Also you can find documentation [in runner-images repo](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md) with guideline for manual image generation.

**Comment by kdimail** [comment]
Hello Sergei.
Thanks for your reply.
I got this error:

PS T:\Packer\runner-images\images\windows\templates> .\packer.exe build .\windows-2019.pkr.hcl
Error: 2 error(s) occurred:

* A capture_name_prefix, managed_image_resource_group_name or shared_image_gallery_destination must be specified
* Specify either a location to create the resource group in or an existing build_resource_group_name, but not both.

  on .\windows-2019.pkr.hcl line 151:
  (source code not available)


In the documentation, I see an example for the CI-CD pipeline, but I need some simple commands with vars like subscription. Could you provide additional info for my case? I see no commands connected to my particular case. Thanks.
